src/components/common/TmSelectNetwork.vue

```diff
@@ -46,37 +46,37 @@ export default {
       }
     }
   },
-  mounted: async function() {
-    await this.$apollo.queries.networks.refetch()
-    if (this.networks.length > 0) {
-      this.sortedNetworks = [
-        // current network first
-        this.networks.find(({ id }) => id === this.networkId),
-        ...this.networks
-          // ignore the current network in the rest of the list as already showing on the top
-          .filter(({ id }) => id !== this.networkId)
-          // show all mainnets next
-          .sort((a, b) => {
-            return a.testnet - b.testnet
-          })
-          // show the default network on the top
-          .sort((a, b) => {
-            return b.default - a.default
-          })
-      ]
-        // filter out undefineds (happens if this.networkId is not set like in the extension)
-        .filter(x => !!x)
-    } else {
-      this.sortedNetworks = []
-    }
-  },
   methods: {
     async selectNetworkHandler(network) {
       if (this.networkId !== network.id) {
         this.$store.dispatch(`setNetwork`, network)
       }
 
       this.$router.push(this.whichFlow)
+    },
+    updateSelectedNetwork(networks) {
+      if (networks.length > 0) {
+        this.sortedNetworks = [
+          // current network first
+          networks.find(({ id }) => id === this.networkId),
+          ...networks
+            // ignore the current network in the rest of the list as already showing on the top
+            .filter(({ id }) => id !== this.networkId)
+            // show all mainnets next
+            .sort((a, b) => {
+              return a.testnet - b.testnet
+            })
+            // show the default network on the top
+            .sort((a, b) => {
+              return b.default - a.default
+            })
+        ]
+          // filter out undefineds (happens if this.networkId is not set like in the extension)
+          .filter(x => !!x)
+      } else {
+        this.sortedNetworks = []
+      }
+      return this.sortedNetworks // just for tests
```

Comment:
instead you can check how `self.sortedNetworks` looks after running this function